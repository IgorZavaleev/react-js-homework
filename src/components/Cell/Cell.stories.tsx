import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Cell } from "./Cell";

export default {
  title: "Cell",
  component: Cell,
} as ComponentMeta<typeof Cell>;

const Template: ComponentStory<typeof Cell> = (args) => <Cell {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  coords: [1, 1],
};
