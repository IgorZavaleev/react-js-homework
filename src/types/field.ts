export type Coords = [number, number];

export type CellProps = {
  coords: Coords;
  state: CellState;
  onClick: onClickType;
  size: GameSize;
};

export type GameSize = "small" | "medium" | "big";
export type Density = "low" | "medium" | "high";
export type GameSpeed = "pause" | "slow" | "medium" | "fast";

export type CellState = 0 | 1;

export type FieldType = CellState[][];
export type onClickType = (coords: Coords) => void;

export type GameState = {
  field: FieldType;
  onClick: onClickType;
  setGameSize: (gameSize: GameSize) => void;
  gameSize: GameSize;
  setGameSpeed: (gameSpeed: GameSpeed) => void;
  gameSpeed: GameSpeed;
  generationNumber: number;
  setDensity: (density: Density) => void;
  clearField: () => void;
};
