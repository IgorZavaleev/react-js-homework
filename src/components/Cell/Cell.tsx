import React from "react";
import { FC } from "react";
import { CellProps } from "../../types/field";

export const Cell: FC<CellProps> = ({ coords, state, onClick }) => (
  <div
    css={{
      backgroundColor: state ? "green" : "red",
      width: 50,
      height: 50,
      border: "solid",
      borderWidth: 1,
    }}
    onClick={() => onClick(coords)}
    data-testid={`${coords}`}
  >
    {`${coords}`}
  </div>
);
