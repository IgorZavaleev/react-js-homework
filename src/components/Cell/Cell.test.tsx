import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { Cell } from "./Cell";
import { CellProps, Coords } from "@/types/field";

describe("Cell component check", () => {
  const coords: Coords = [1, 1];

  const props: CellProps = {
    coords: coords,
    state: 1,
    onClick: jest.fn(),
    size: "small",
  };

  it("Cell renders correctly", () => {
    const { asFragment } = render(<Cell {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("onClick should be called", () => {
    render(<Cell {...props} />);

    const cellComp = screen.getByTestId(`cell_${coords[0]}_${coords[1]}_alive`);
    fireEvent.click(cellComp);

    expect(props.onClick).toBeCalled();
  });
});
