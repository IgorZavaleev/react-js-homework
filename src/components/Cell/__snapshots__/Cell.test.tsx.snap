// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component check Cell renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-eudiny-Cell"
    data-testid="1,1"
  >
    1,1
  </div>
</DocumentFragment>
`;
